Reorganize codelab (#352)
transparency-dev/trillian-tessera: cmd/conformance/mysql/README.md

# Conformance MySQL log

> This binary runs an HTTP web server that accepts POST HTTP requests to a `/add` endpoint.

Reviewer: ```suggestion
This binary runs an HTTP web server that accepts POST HTTP requests to an `/add` endpoint.
```

This endpoint takes arbitrary data and adds it to a MySQL based Tessera log.

> [!WARNING]
> - This is an example and is not fit for production use, but demonstrates a way of using the Tessera Log with MySQL storage backend.
> - This example is built on the [tlog tiles API](https://c2sp.org/tlog-tiles) for read endpoints and exposes a /add endpoint that allows any POSTed data to be added to the log.

## Bring up a log

This will help you bring up a MySQL database to store a Tessera log, and start a personality
binary that can add entries to it.

> You can either start this log using Docker Compose, or manually with `go run`.

Reviewer: ```suggestion
You can run this personality using Docker Compose or manually with `go run`.
```

Note that all the commands are executed at the root directory of this repository.

### Docker Compose

#### Prerequisites

Install [Docker Compose](https://docs.docker.com/compose/install/).

#### Start the log

```sh
docker compose -f ./cmd/conformance/mysql/docker/compose.yaml up
```

Add `-d` if you want to run the log in detached mode.

#### Stop the log

```sh
docker compose -f ./cmd/conformance/mysql/docker/compose.yaml down
```

### Manual 

#### Prerequisites

> You need to have a MySQL database running on port `3306`.

Reviewer: Will the reader also need to have configured the MySQL instance to accept password auth for `root` and set the password to `root`, and created a DB instance called `test_tessera`?

Author: @roger2hk

Reviewer 2: Yes, the "starting the log" command in the next few lines requires the user and password to be `root`. We can turn them into variables, but I think it's easier to define them for easy & paste.

Reviewer 2: If we don't want to use the `root` user, we can define the `MYSQL_DATABASE`, `MYSQL_USER`, `MYSQL_PASSWORD` in the `docker run` command.

Author: Alright, added some instructions.

You can start one using [Docker](https://docs.docker.com/engine/install/).

```sh
docker run --name test-mysql -p 3306:3306 -e MYSQL_ROOT_PASSWORD=root -e MYSQL_DATABASE=test_tessera -d mysql:8.4
```

#### Start the log

```sh
go run ./cmd/conformance/mysql --mysql_uri="root:root@tcp(localhost:3306)/test_tessera" --init_schema_path="./storage/mysql/schema.sql" --private_key_path="./cmd/conformance/mysql/docker/testdata/key"
```

#### Stop the log

<kbd>Ctrl</kbd> <kbd>C</kbd>

## Add entries to the log

### Manually

> Head over to the [codelab](../#codelab) to manually add entries to the log, and read them.

Reviewer: ```suggestion
Head over to the [codelab](../#codelab) to manually add entries to the log, and read them back.
```

Author: used "and inspect the log" to be more accurate.
